src/features/api-keys/lib.ts

```diff
@@ -278,8 +280,8 @@ export const createGenericApiKey = async (
 };
 
 export const isApiKeyWithRole = (() => {
-	const authQuery = _.once(() =>
-		api.Auth.prepare<{ key: string; roleName: string }, 'role'>({
+	const authQuery = _.once(() => {
+		const opts = {
```

Comment:
I would prefer this to be named `params` as it directly maps to the `Params<T>` typing and `opts` might be confusing with the inner `options`